pkg/auth/cred.go

```diff
@@ -33,6 +37,48 @@ type expireEarlyTokenCredential struct {
 	cred azcore.TokenCredential
 }
 
+func GetAuxiliaryToken(ctx context.Context) (azcore.AccessToken, error) {
```

Comment:
This code is not covered by any unit tests. Ideally we would have an acceptance unit test that checks that during provisioning calls to create VM carry additional token in SIG scenario, but unfortunately our current fakes are not flexible enough for this. More focused unit testing of just GetAuxilaryToken with something like a mock server would be second best. If we have E2E tests that cover these scenarios - this becomes less critical, and can be added later.